core/trino-main/src/main/java/io/trino/operator/DeleteAndInsertMergeProcessor.java

```diff
@@ -111,7 +113,7 @@ public Page transformPage(Page inputPage)
                 case DEFAULT_CASE_OPERATION_NUMBER -> { /* ignored */ }
                 case INSERT_OPERATION_NUMBER -> insertPositions++;
                 case DELETE_OPERATION_NUMBER -> deletePositions++;
-                case UPDATE_OPERATION_NUMBER -> updatePositions++;
+                case UPDATE_OPERATION_NUMBER, UPDATE_INSERT_OPERATION_NUMBER, UPDATE_DELETE_OPERATION_NUMBER -> updatePositions++;
```

Comment:
actually, is UPDATE_INSERT_OPERATION_NUMBER, UPDATE_DELETE_OPERATION_NUMBER case reachable here?
we're in DeleteAndInsertMergeProcessor class and it seems this the class responsible for converting UPDATE rows into INSERT/DELETE rows, so insert-from-an-update and delete-from-an-update do not exist yet

Reply:
```
// This class will create such rows, they are not expected on input 
                case UPDATE_INSERT_OPERATION_NUMBER, UPDATE_DELETE_OPERATION_NUMBER -> throw new IllegalArgumentException("Unexpected operator number: " + operation);
```